raiden/api/python.py

```diff
@@ -1167,7 +1165,10 @@ def get_raiden_events_payment_history_with_timestamps(
                 token_address=token_address,
             )
         ]
-
+        if offset:
+            events = events[offset:]
+        if limit:
+            events = events[:limit]
```

Comment:
This is not good for performance, instead of loading every event from the database and then filtering, the code should do the opposite, and filter the events before returning them. (Note: I assume that with SQLite and our current DB modelling this is not as big of an issue, but it is better to fix it instead of waiting for it to show as a problem in production).

To do that the `SQLiteStorage::get_raiden_events_with_timestamps` needs three new arguments `event_types`, `token_address`, and `partner_address`, these arguments should be optional, and a different query should be used depending on the filters provided. When all three arguments are provided the SQL will be roughly:

```sql
SELECT
  data, timestamp
FROM
  state_events
WHERE
  json_extract(data, '_field') IN (?) AND -- needs to test if the ? placeholder can be a list of values
  (
    json_extract(data, 'initiator') == ? OR
    json_extract(data, 'target')  == ?
  )
```